x/concentrated-liquidity/tick_test.go

```diff
@@ -353,3 +353,95 @@ func (s *KeeperTestSuite) TestGetTickInfo() {
 		})
 	}
 }
+
+func (s *KeeperTestSuite) TestCrossTick() {
+	var (
+		preInitializedTickIndex = DefaultCurrTick.Int64() - 2
+	)
+
+	tests := []struct {
+		name      string
+		poolToGet uint64
+		tickToGet int64
+		// upon setting this to true, we manipulate tick's fee growth
+		// to a value that is greater than that of accumulator
+		manipulateTickFeeGrowth      bool
+		expectedLiquidityDelta       sdk.Dec
+		expectedTickFeeGrowthOutside sdk.DecCoins
+		expectedErr                  bool
+	}{
+		{
+			name:                         "Get tick info on existing pool and existing tick",
+			poolToGet:                    validPoolId,
+			tickToGet:                    preInitializedTickIndex,
+			expectedLiquidityDelta:       DefaultLiquidityAmt.Neg(),
+			expectedTickFeeGrowthOutside: DefaultFeeAccumCoins,
+		},
+		{
+			name:        "Try invalid tick",
+			poolToGet:   2,
+			tickToGet:   preInitializedTickIndex,
+			expectedErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		s.Run(test.name, func() {
+			// Init suite for each test.
+			s.Setup()
+
+			// Create a default CL pool
+			s.PrepareConcentratedPool()
+
+			if test.poolToGet == validPoolId {
+				s.FundAcc(s.TestAccs[0], sdk.NewCoins(sdk.NewCoin("ETH", sdk.NewInt(10000000000000)), sdk.NewCoin("USDC", sdk.NewInt(1000000000000))))
+				s.SetupPosition(test.poolToGet, s.TestAccs[0], DefaultCoin0, DefaultCoin1, DefaultLowerTick, DefaultUpperTick)
+			}
+
+			// Charge fee to make sure that the global fee accumulator is always updated.
+			// This is to test that the per-tick fee growth accumulator gets initialized.
+			defaultAccumCoins := sdk.NewDecCoin("foo", sdk.NewInt(50))
+			err := s.App.ConcentratedLiquidityKeeper.ChargeFee(s.Ctx, validPoolId, defaultAccumCoins)
+			s.Require().NoError(err)
+
+			// Set up an initialized tick
+			err = s.App.ConcentratedLiquidityKeeper.InitOrUpdateTick(s.Ctx, validPoolId, preInitializedTickIndex, DefaultLiquidityAmt, true)
+			s.Require().NoError(err)
+
+			// update the fee accumulator so that we have accum value > tick fee growth value
+			// now we have 100 foo coins inside the pool accumulator
+			err = s.App.ConcentratedLiquidityKeeper.ChargeFee(s.Ctx, validPoolId, defaultAccumCoins)
+			s.Require().NoError(err)
+
+			// manipulate tick fee growth
+			if test.manipulateTickFeeGrowth {
+				err = s.App.ConcentratedLiquidityKeeper.ChargeFee(s.Ctx, validPoolId, defaultAccumCoins)
+				s.Require().NoError(err)
+
+				err = s.App.ConcentratedLiquidityKeeper.InitOrUpdateTick(s.Ctx, validPoolId, preInitializedTickIndex, DefaultLiquidityAmt, true)
+				s.Require().NoError(err)
```

Comment:
What is the difference between the manipulation/charge fee above and here?

Reply:
Deleted the manipulation / charge fee logic!